include/boost/multiprecision/tommath.hpp

```diff
@@ -245,10 +243,8 @@ struct tommath_int
          return *this;
       }
 
-      #if !defined(BOOST_MP_STANDALONE) || defined(BOOST_MATH_STANDALONE)
-      BOOST_MP_ASSERT(!(boost::math::isinf)(a));
-      BOOST_MP_ASSERT(!(boost::math::isnan)(a));
-      #endif
+      BOOST_MP_ASSERT(!(boost::multiprecision::detail::isinf)(a));
```

Comment:
Another where an unqualified call and update to BOOST_MP_FLOAT128_USING would fix things.